core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceStorageStatusVerifier.kt

```diff
@@ -0,0 +1,182 @@
+    private val fileResourceNetworkHandler: FileResourceNetworkHandler,
+    private val coroutineAPICallExecutor: CoroutineAPICallExecutor,
+) {
+
+    /**
+     * Verifies the storage status of multiple file resources in batch.
+     *
+     * @param fileResourceUids List of file resource UIDs to verify
+     * @param maxAttempts Maximum number of polling attempts per file (default: 30)
+     * @param delayMs Delay in milliseconds between polling attempts (default: 1000ms)
+     * @param batchSize Number of files to verify in parallel (default: 10)
+     * @return Map of file resource UID to verification result (true if STORED, false otherwise)
+     */
+    suspend fun verifyStorageStatusBatch(
+        fileResourceUids: List<String>,
+        maxAttempts: Int = DEFAULT_MAX_ATTEMPTS,
+        delayMs: Long = DEFAULT_DELAY_MS,
+        batchSize: Int = DEFAULT_BATCH_SIZE,
+    ): Map<String, FileResourceVerificationResult> {
+        if (fileResourceUids.isEmpty()) {
+            return emptyMap()
+        }
+
+        Log.d(TAG, "Starting batch verification for ${fileResourceUids.size} file resources")
+
+        val results = mutableMapOf<String, FileResourceVerificationResult>()
+        val pendingUids = fileResourceUids.toMutableSet()
+
+        for (attempt in 1..maxAttempts) {
+            if (pendingUids.isEmpty()) {
+                break
+            }
+
+            Log.d(TAG, "Verification attempt $attempt/$maxAttempts for ${pendingUids.size} pending files")
+
+            // Process in batches to avoid overwhelming the server
+            val currentBatch = pendingUids.take(batchSize)
+            val batchResults = verifyBatch(currentBatch)
+
+            batchResults.forEach { (uid, result) ->
+                when (result.status) {
+                    FileResourceStorageStatus.STORED -> {
+                        results[uid] = result
+                        pendingUids.remove(uid)
+                        Log.d(TAG, "File resource $uid is STORED")
+                    }
+                    FileResourceStorageStatus.FAILED -> {
+                        results[uid] = result
+                        pendingUids.remove(uid)
+                        Log.w(TAG, "File resource $uid FAILED to store")
+                    }
+                    FileResourceStorageStatus.PENDING -> {
+                        // Keep in pending list for next attempt
+                        Log.d(TAG, "File resource $uid still PENDING")
+                    }
+                    FileResourceStorageStatus.NONE -> {
+                        // Treat as pending
+                        Log.d(TAG, "File resource $uid has status NONE")
+                    }
+                }
+            }
+
+            // If there are still pending files and we haven't reached max attempts, wait before next poll
+            if (pendingUids.isNotEmpty() && attempt < maxAttempts) {
+                delay((delayMs * (2.0.pow(attempt) - 1)).toLong())
```

Comment:
I think there's a critical bug in the exponential backoff calculation:

`delay((delayMs * (2.0.pow(attempt) - 1)).toLong())`

This formula grows exponentially without bounds:
- Attempt 10: ~17 minutes
- Attempt 20: ~12 days
- Attempt 30: ~34 years

With DEFAULT_MAX_ATTEMPTS = 30, this will cause the app to freeze indefinitely when files fail verification.

Maybe we could do a Linear backoff `delay(delayMs * attempt)`or cap the exponential backoff with a maxDelay like this:

```
val maxDelay = 10_000L
delay(minOf(delayMs * (2.0.pow(attempt - 1)).toLong(), maxDelay))
```